modin/core/dataframe/pandas/dataframe/protocol/column.py

```diff
@@ -0,0 +1,590 @@
+        _k = DTypeKind
+        _np_kinds = {
+            "i": _k.INT,
+            "u": _k.UINT,
+            "f": _k.FLOAT,
+            "b": _k.BOOL,
+            "U": _k.STRING,
+            "M": _k.DATETIME,
+            "m": _k.DATETIME,
+        }
+        kind = _np_kinds.get(dtype.kind, None)
+        if kind is None:
+            # Not a NumPy dtype. Check if it's a categorical maybe
+            if isinstance(dtype, pd.CategoricalDtype):
+                # 23 matches CATEGORICAL type in DTypeKind
+                kind = 23
+            else:
+                raise ValueError(
+                    f"Data type {dtype} not supported by exchange protocol"
+                )
+
+        if kind not in (_k.INT, _k.UINT, _k.FLOAT, _k.BOOL, _k.CATEGORICAL, _k.STRING):
+            raise NotImplementedError(f"Data type {dtype} not handled yet")
+
+        bitwidth = dtype.itemsize * 8
+        format_str = dtype.str
+        endianness = dtype.byteorder if not kind == _k.CATEGORICAL else "="
+        return (kind, bitwidth, format_str, endianness)
+
+    @property
+    def describe_categorical(self) -> Dict[str, Any]:
+        """
+        If the dtype is categorical, there are two options:
+        - There are only values in the data buffer.
+        - There is a separate dictionary-style encoding for categorical values.
+
+        TBD: are there any other in-memory representations that are needed?
+
+        Returns
+        -------
+        dict
+            Content of returned dict:
+            - "is_ordered" : bool, whether the ordering of dictionary indices is
+                             semantically meaningful.
+            - "is_dictionary" : bool, whether a dictionary-style mapping of
+                                categorical values to other objects exists
+            - "mapping" : dict, Python-level only (e.g. ``{int: str}``).
+                          None if not a dictionary-style categorical.
+
+        Raises
+        ------
+        ``RuntimeError`` if the dtype is not categorical.
+        """
+        if not self.dtype[0] == DTypeKind.CATEGORICAL:
+            raise TypeError(
+                "`describe_categorical only works on a column with "
+                "categorical dtype!"
+            )
+
+        # TODO: Raise an exception if ``self._allow_copy==False``?
+        pandas_series = self._col.to_pandas().squeeze(axis=1)
+        ordered = pandas_series.dtype.ordered
+        is_dictionary = True
+        # NOTE: this shows the children approach is better, transforming
+        # `categories` to a "mapping" dict is inefficient
+        # codes = self._col.values.codes  # ndarray, length `self.size`
+        # categories.values is ndarray of length n_categories
+        # TODO: Raise an exception if ``self._allow_copy==False``?
+        categories = pandas_series.values.categories.values
+        mapping = {ix: val for ix, val in enumerate(categories)}
+        return ordered, is_dictionary, mapping
```

Comment:
the method has to return a dictionary, not a tuple

Reply:
fixed